packages/nuxt/src/plugins/plugin.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  CreateDependencies,
+  CreateNodes,
+  CreateNodesContext,
+  TargetConfiguration,
+  detectPackageManager,
+  getPackageManagerCommand,
+  joinPathFragments,
+  offsetFromRoot,
+  readJsonFile,
+  writeJsonFile,
+} from '@nx/devkit';
+import { basename, dirname, join } from 'path';
+import { projectGraphCacheDirectory } from 'nx/src/utils/cache-directory';
+import { readTargetDefaultsForTarget } from 'nx/src/project-graph/utils/project-configuration-utils';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+import { existsSync, readdirSync } from 'fs';
+import { loadNuxtKitDynamicImport } from '../utils/executor-utils';
+import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
+
+const cachePath = join(projectGraphCacheDirectory, 'nuxt.hash');
+const targetsCache = existsSync(cachePath) ? readTargetsCache() : {};
+
+const calculatedTargets: Record<
+  string,
+  Record<string, TargetConfiguration>
+> = {};
+
+function readTargetsCache(): Record<
+  string,
+  Record<string, TargetConfiguration>
+> {
+  return readJsonFile(cachePath);
+}
+
+function writeTargetsToCache(
+  targets: Record<string, Record<string, TargetConfiguration>>
+) {
+  writeJsonFile(cachePath, targets);
+}
+
+export const createDependencies: CreateDependencies = () => {
+  writeTargetsToCache(calculatedTargets);
+  return [];
+};
+
+export interface NuxtPluginOptions {
+  buildTargetName?: string;
+  serveTargetName?: string;
+  testTargetName?: string;
+}
+
+export const createNodes: CreateNodes<NuxtPluginOptions> = [
+  '**/nuxt.config.{js,ts}',
+  async (configFilePath, options, context) => {
+    const projectRoot = dirname(configFilePath);
+    // Do not create a project if package.json and project.json isn't there.
+    const siblingFiles = readdirSync(join(context.workspaceRoot, projectRoot));
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    options = normalizeOptions(options);
+
+    const hash = calculateHashForCreateNodes(projectRoot, options, context);
```

Comment:
Also add the lockfile into the hash.

See: https://github.com/nrwl/nx/blob/master/packages/cypress/src/plugins/plugin.ts#L75